### DISCUSSION TITLE
[v3] @entangle upgrade routine improperly inserting and placing .defer

### DISCUSSION THREAD

**Discussion by markjaquith** [livewire/livewire#5868] [7. Bugs (archived)]
[Jetstream's Modal template][template] is upgraded improperly by the automatic upgrader:

```diff
 <div
     x-data="{
-        show: @entangle($attributes->wire('model')).defer,
+        show: @entangle($attributes->wire('model').live).defer,
         focusables() {
```

[template]: https://github.com/laravel/jetstream/blob/51a9ebf97839d2ae6e39ec8eb22b0555b1db20e3/stubs/livewire/resources/views/components/modal.blade.php#L16

1. Not valid syntax.
2. Since this was `.defer`, it should not be getting `.live` at all.

I suspect it's the parens from the `$attributes->wire('model')` call messing it up.

**Comment by joshhanley** [comment]
@markjaquith thanks for reporting! @PhiloNL FYI, if I get a chance I will have a look to see if I can work out what is going on 🙂

**Reply by PhiloNL** [reply]
Thanks! I'll take a look as well. It's probably related to the regex looking for the first closing `)` instead of the last.

**Reply by joshhanley** [reply]
@PhiloNL ah yep! Ok thanks, much appreciated

**Reply by erikscheepers** [reply]
I had also an issue with the entangle upgrade routine (don't know if this is already resolved by a later update). But if you're still looking into that regex: keep in mind that just looking for the latest closing `)` is too easy:
```diff
<div x-data="{
-     customer_table_search: $persist(@entangle('search')).live
+     customer_table_search: $persist(@entangle('search').live)
}"></div>```

**Reply by livelogic** [reply]
The upgrader in v3.0.1 also does another bad pattern replacement for `$attributes->whereStartsWith('wire:model')` where you don't expect any modification.

**Reply by hussain-alkhatib** [reply]
sory i have problem with that 
in v2 when user reload page it was take value from local storage and send it to component 
but in v3 it get default value frome component and set it local storage

i already update my code to :

```
<div class="col-12 col-md-6 col-lg-3 mb-2 mb-lg-0"
        x-data="{
            search: $persist(@entangle('search').live),
        }" >
        <input type="search" placeholder="search" class="form-control" wire:model.live="search">
</div>
```
so how can i fix that
